scripts/setup.sh

```diff
@@ -190,3 +202,15 @@ if [ ! -d "${CONFIG_DIR}" ] ; then
 else
     echo "Configuration directory '${CONFIG_DIR}' exists, not overwriting"
 fi
+
+# TODO: What if proxies already set via env
+# update the config/group_vars/all.yml file for proxy config
+if [ `grep -v ^# ${SCRIPT_DIR}/proxy.sh | grep -v ^$ | wc -l` -gt 0 ]; then
+    o=config.example/group_vars/all.yml
+    t=config/group_vars/all.yml 
+    echo "Updating [$t] to use proxies"
+    . ${SCRIPT_DIR}/proxy.sh && cat $o | sed -e "s|#PROXY_ENV|proxy_env:|g" \
+	                                    -e "s|#NO_PROXY|no_proxy: $no_proxy|g" \
+	                                    -e "s|#HTTP_PROXY|http_proxy: $http_proxy|g" \
+	                                    -e "s|#HTTPS_PROXY|https_proxy: $https_proxy|g" > $t
+fi
```

Comment:
As in my comment for `config.example/group_vars/all.yml`: This seems pretty fragile, and will break if the user accidentally changes the comments in the config file or uses a different config path. The config directory is intended to be user-editable, so this seems likely to happen not infrequently.

Instead, we should just ask the user to configure proxies in both the environment shell script and in the Ansible group vars. It represents some duplication, but I think it's less likely to break, especially since proxies are usually configured once and then left alone.

Reply:
best to keep its as simple as possible and the reason for keeping edits to a single file thus using proxy.sh. the setup.sh script installs software and creates the initial contents of config dir, asking users to edits in multiple locations leads to more confusion rather than less. it's the same as having to edit the inventory file to make the config applicable to their env. In this case if they want difference proxies for setup.sh and software install from setup.sh and the ansible scripts they'll need to update group_vars. The likelihood of that in an enterprise is extremely low.